core/src/main/scala/chisel3/internal/MonoConnect.scala

```diff
@@ -411,6 +412,7 @@ private[chisel3] object MonoConnect {
     source:     Property[_],
     context:    BaseModule
   ): Unit = {
+    checkConnect.checkPropertyConnection(sourceInfo, sink, source)
```

Comment:
The fix for `checkConnect.apply` on Property is here: https://github.com/chipsalliance/chisel/pull/3635. Once that is in, I will rebase this and make sure here we just call `checkConnect`, and the Property version of `checkConnect` calls `checkPropertyConnection`.